yarn-project/acir-simulator/src/client/private_execution.ts

```diff
@@ -218,12 +218,9 @@ export class PrivateFunctionExecution {
 
         const notePreimage = new NotePreimage(preimage);
         const txAuxData = new TxAuxData(notePreimage, contractAddress, storageSlot);
-
-        const owner = {
-          x: fromACVMField(ownerX),
-          y: fromACVMField(ownerY),
-        };
-        const ownerPublicKey = Point.fromBuffer(Buffer.concat([owner.x.toBuffer(), owner.y.toBuffer()]));
+        const ownerPublicKey = new Point(
+          Buffer.concat([fromACVMField(ownerX).toBuffer(), fromACVMField(ownerY).toBuffer()]),
```

Comment:
this still does the same thing though, `fromACVMField` converts from string -> Buffer -> Field that we then convert `toBuffer()`.

Could have something like a little helper that does:
```
toFieldBuffer(value: string) {
  return Buffer.from(value.slice(2), 'hex').subarray(0, Fr.SIZE_IN_BYTES)
}
```

Reply:
think this might be useful in other parts too, maybe it could be static method in `Field`?

Reply:
Addressed again in [4e4795d](https://github.com/AztecProtocol/aztec-packages/pull/777/commits/4e4795d4179393a45853f42fd5fba79da08b6cb7).

Looks good now?